script.js

```diff
@@ -1,12 +1,421 @@
+      let title = element.textContent;
+
+      let appearsOnText = title.toLowerCase().indexOf(searchValue) != -1;
+      console.log();
+      // console.log(element.className === `container episode`);
+      if (
+        appearsOnText &&
+        element.className === `container show` &&
+        (element.className === `container episode`) === false
+      ) {
+        element.style.display = "initial";
+      } else if (
+        appearsOnText &&
+        ((element.className === `container show`) === true).length +
+          ((element.className === `container show`) === false).length >
+          ((element.className === `container show`) === true).length
+        // &&  (element.className === `container episode`) === true
+      ) {
+        // if (
+        //   appearsOnText &&
+        //   element.className === `container show` &&
+        //   (element.className === `container episode`) === true
+        // ) {
+        //   element.style.display = "initial";
+        // } else {
+        //   element.style.display = "none";
+        // }
+        console.log(element.className === `container episode`);
+      } else {
+        element.style.display = "none";
+      }
+    });
+  });
+  // result();
+}
+
+//counter result
+function result() {
+  let resultDiv = document.createElement(`div`);
+  header.appendChild(resultDiv);
+  resultDiv.setAttribute(`id`, `resultDiv`);
+  resultDiv.setAttribute(`class`, `resultDiv`);
+  let resultParagraph = document.createElement(`p`);
+  resultDiv.appendChild(resultParagraph);
+  resultParagraph.setAttribute(`id`, `result`);
+  resultParagraph.setAttribute(`class`, `result`);
+  resultDiv.setAttribute(`title`, `Result / Total`);
+  let containerDiv = document.getElementsByClassName(`container`);
+  let container = Array.from(containerDiv);
+  let searchBar = document.getElementById(`searchInput`);
+
+  //main function of counting after and before keyup
+  resultParagraph.innerHTML = `${codeCorrection(
+    container.length
+  )} / ${codeCorrection(container.length)}`;
+  searchBar.addEventListener(`keyup`, () => {
+    resultParagraph.innerHTML = `${codeCorrection(
+      container.filter((x) => x.style.display === `initial`).length
+    )} / ${codeCorrection(container.length)}`;
+  });
+}
+
+//for up date result out of searchbar
+function updateResult() {
+  setTimeout(function () {
+    let containerDiv = document.getElementsByClassName(`container`);
+    let container = Array.from(containerDiv);
+    let resultParagraph = document.getElementById(`result`);
+    // console.log(container.filter((x) => x.style.display != `none`));
+    // console.log(container);
+    resultParagraph.innerHTML = `${codeCorrection(
+      container.filter((x) => x.style.cssText === `display: initial;`).length
+    )} / ${codeCorrection(
+      container.filter((x) => x.style.cssText !== `display: none;`).length
+    )}`;
+    // console.log(container.filter((x) => x.style.cssText === `display: initial;`));
+  }, 400);
```

Comment:
setTimeout here is not a good solution. On a slow internet connection this will break! I think you should find a way to remove this.

Hint: the solution will involve thinking more about the `.then()` keyword ;)

Reply:
oh, many thanks for you time to read and all of these amazing comments.